control/autoware_autonomous_emergency_braking/src/node.cpp

```diff
@@ -348,21 +348,27 @@ bool AEB::fetchLatestData()
     return missing("object detection method (pointcloud or predicted objects)");
   }
 
-  const auto imu_ptr = sub_imu_.takeData();
-  if (use_imu_path_) {
+  const bool has_imu_path = std::invoke([&]() {
+    if (!use_imu_path_) return false;
+    const auto imu_ptr = sub_imu_.takeData();
     if (!imu_ptr) {
       return missing("imu message");
     }
     // imu_ptr is valid
     onImu(imu_ptr);
-  }
-  if (use_imu_path_ && !angular_velocity_ptr_) {
-    return missing("imu");
-  }
+    return (!angular_velocity_ptr_) ? missing("imu") : true;
+  });
 
-  predicted_traj_ptr_ = sub_predicted_traj_.takeData();
-  if (use_predicted_trajectory_ && !predicted_traj_ptr_) {
-    return missing("control predicted trajectory");
+  const bool has_predicted_path = std::invoke([&]() {
+    if (!use_predicted_trajectory_) {
+      return false;
+    }
+    predicted_traj_ptr_ = sub_predicted_traj_.takeData();
+    return (!predicted_traj_ptr_) ? missing("control predicted trajectory") : true;
+  });
+
+  if (!has_imu_path && !has_predicted_path) {
+    return missing("any type of path");
```

Comment:
```suggestion
    if (use_imu_path_ && use_predicted_trajectory_) {
      RCLCPP_INFO_SKIPFIRST_THROTTLE(
        get_logger(), *get_clock(), 5000,
        "[AEB] At least one path (IMU or predicted trajectory) is required for operation");
    }
    return false;
```
@danielsanchezaran 
Isn't it sufficient to output "At least one path is required" only when both `use_imu_path_` and `use_predicted_trajectory_` are true?

Reply:
@kyoichi-sugahara this check "(use_imu_path_ && use_predicted_trajectory_)" is not really necessary, the AEB requires at least one path (imu or predicted_path) so the lack of at least 1 path should be enough to trigger the message, in the case that, for example the user sets use_imu_path_ to true and use_predicted_trajectory_ to false and there is NO Imu topic, I believe printing this message is not a problem either. I have modified the code to reflect this